thriftserver/thriftserver/base/params.env

```diff
@@ -0,0 +1,6 @@
+storage_class=
+s3_endpoint=
```

Comment:
It is missing the variables for the S3 keys.

Reply:
@rimolive I don't think we should be putting those in here at all. If we do that, the S3 credentials would end up in a plaintext as `parameters` in the `kfdef` and in the end they will end up in a `ConfigMap`. And that's not a safe way to treat credentials. Sensitive data should be stored in `Secrets` only. It's the same as we do for the DB credentials - we inline the defaults in the secret. See Jupyter Hub component for example. It's up to the used to override those. I know kfctl can [handle secrets](https://github.com/kubeflow/kfctl/blob/master/pkg/apis/apps/kfdef/v1/README.md#kfdefspecsecrets) and load them from the local environment (via `secrets.env`), but I don't think that applies to the operator scenario. Do you know about any sensible solution?

Reply:
How are we expecting ODH users to provide these values? We should have an opinionated, documented, recommendation. I think we can all agree that we don't want them in plain text somewhere, and they probably shouldn't be defined in the ODH (other than values like "CHANGEME") since they'll be unique to each deployment.

Reply:
@tumido I think it would be a great contribution from ODH to KF operator if we could make the secrets in KFDef actually useful - i.e. instead of pulling the values from env vars, they could be pulled from actual secrets imported to the cluster by the user. WDYT?

Reply:
> How are we expecting ODH users to provide these values? We should have an opinionated, documented, recommendation. I think we can all agree that we don't want them in plain text somewhere, and they probably shouldn't be defined in the ODH (other than values like "CHANGEME") since they'll be unique to each deployment.

My personal approach to this is:
1. Avoid using parameters as much as possible
2. If something is unique per deployment AND is NOT a secret, put it in parameter
3. If something is unique per deployment AND IS secret, put it in secret with `CHANGME` and recommend http://opendatahub.io/docs/administration/installation-customization/customization.html
4. Is something can have a sane value which only need to be changed in specific (e.g. production) deployments, put it in CM/Secret and recommend http://opendatahub.io/docs/administration/installation-customization/customization.html

Reply:
> I know kfctl can [handle secrets](https://github.com/kubeflow/kfctl/blob/master/pkg/apis/apps/kfdef/v1/README.md#kfdefspecsecrets) and load them from the local environment (via `secrets.env`)

FYI I have spent some time digging through the code to figure out how these secrets work and I found how/where they are loaded (https://github.com/kubeflow/kfctl/blob/master/pkg/kfconfig/loaders/v1.go#L111-L130), but I could not find a single location where it would be used:D

I also found this issue: https://github.com/kubeflow/kfctl/issues/308 which supports my hunch that there is some code missing:)

My goal was to find how it's used and propose extension where the `SecretSource` could also be an actual Secret in the cluster

Reply:
Nevermind..found it: https://github.com/kubeflow/kfctl/blob/master/pkg/kfapp/gcp/gcp.go#L183

It is used by plugins to configure platform specific values (like the example above GCP Access Token Name).

So this does not seem like something we could use, unless we want to write an OpenShift plugin...I believe these secrets are not what we want